packages/vue-app/template/mixins/fetch.client.js

```diff
@@ -0,0 +1,53 @@
+import Vue from 'vue'
+import { hasFetch } from '../utils'
+
+const isSsrHydration = (vm) => vm.$vnode && vm.$vnode.elm && vm.$vnode.elm.dataset && vm.$vnode.elm.dataset.ssrKey
+const nuxtState = window.<%= globals.context %>
+
+export default {
+  beforeCreate () {
+    if (hasFetch(this)) {
+      Vue.util.defineReactive(this, '$isFetching', false)
+      Vue.util.defineReactive(this, '$fetchError', null)
+    }
+  },
+  created () {
+    if (hasFetch(this) && isSsrHydration(this)) {
+      // Hydrate component
+      this._hydrated = true
+      this.$isLoading = false
+      this._lastFetchAt = Date.now()
+
+      this._ssrKey = +this.$vnode.elm.dataset.ssrKey
+      const data = nuxtState.data[this._ssrKey]
+
+      // If fetch error
+      if (data && data._error) {
+        return this.$fetchError = data._error
+      }
+      for (const key in data) {
+        this[key] = data[key]
+      }
+    }
+  },
+  beforeMount () {
+    if (!this._hydrated && hasFetch(this)) {
+      this.$fetch()
+    }
+  },
+  methods: {
+    async '$fetch' () {
+      this.$nuxt.nbFetching++
+      this.$isFetching = true
+      this.$fetchError = null
```

Comment:
@Atinux Likewise here, instead of having `$fetchError`, we could simply have `$fetch.error` or `$fetchState.error` where `$fetchState` would also give access to the other properties.

Reply:
Actually this is just for layouts here to keep track of actual fetch() running